### PR TITLE
feat: add user metadata

### PR DESCRIPTION
**Pull request by muneebkq**
add metadata to user license payload

### CODE REVIEW

File: src/extern_functions.rs [hunk 1]
```diff
@@ -35,16 +36,19 @@ pub type CallbackType = extern "C" fn(LexActivatorCode);
 extern "C" {
     // --------------- Setter Functions ---------------
     pub fn SetProductData(productData: cstrtype!()) -> c_int;
+    pub fn SetProductFile(filePath: cstrtype!()) -> c_int;
```
Comment (ahmad-kemsan) [inline src/extern_functions.rs:39]: @muneebkq Remove this function, as we have deprecated SetProductFile(), so we will not include it in our rust wrapper.

File: src/lib.rs [hunk 2]
```diff
@@ -229,6 +274,28 @@ pub fn set_debug_mode(enable: u32) {
...
+///
+/// single machine and require real-time updates to the license state.
+///
+/// # Arguments
+///
+/// * `mode` - False or True to disable or enable caching.
+///
+/// Returns `Ok(())` if mode is set successfully.
+
+pub fn set_cache_mode(mode: bool) -> c_uint {
```
Comment (ahmad-kemsan) [inline src/lib.rs:294]: @muneebkq Please follow the Result return pattern for set_cache_mode().

**Review by ahmad-kemsan — CHANGES_REQUESTED** [review]
Please check the comments

**Comment by ahmad-kemsan** [comment]
@muneebkq To fix the windows build error, update the version of LexActivator to 3.30.0 in download.sh. It should fix the issue.

File: src/lib.rs (hunk 2 above)
Comment (muneebkq) [reply src/lib.rs:294]: Jsdoc?

File: src/extern_functions.rs (hunk 1 above)
Comment (muneebkq) [reply src/extern_functions.rs]: removed

File: src/lib.rs (hunk 2 above)
Comment (muneebkq) [reply src/lib.rs]: result pattern followed

File: src/lib.rs [hunk 3]
```diff
@@ -433,6 +470,37 @@ pub fn set_trial_activation_metadata(key: String, value: String) -> Result<(), L
...
+
+/// # Arguments
+///
+/// * `app_version` - string of maximum length 256 characters.
+///
+/// # Returns
+///
+/// Returns `Ok(())` if the release version is set successfully, If an error occurs, an `Err` containing the `LexActivatorError`is returned.
+
+pub fn set_app_version(app_version: String) -> Result<(), LexActivatorError> {
```
Comment (ahmad-kemsan) [inline src/lib.rs:485]: @muneebkq There is no need for adding set_app_version(). This function has been deprecated as we already have the set_release_version().

File: src/extern_functions.rs [hunk 4]
```diff
@@ -37,12 +38,13 @@ extern "C" {
     pub fn SetProductData(productData: cstrtype!()) -> c_int;
     pub fn SetProductId(productId: cstrtype!() , flags: c_uint) -> c_int;
     pub fn SetDataDirectory(dataDir: cstrtype!()) -> c_int;
-    pub fn SetDebugMode(enable: c_uint) -> c_uint;
+    pub fn SetDebugMode(enable: c_uint) -> c_int;
+    pub fn SetCacheMode(mode: c_int) -> c_int;
```
Comment (ahmad-kemsan) [inline src/extern_functions.rs:42]: @muneebkq SetCacheMode() take uint as an arg, please refer to https://github.com/cryptlex/lexactivator-c/blob/master/examples/LexActivator.h#L460

**Review by ahmad-kemsan — CHANGES_REQUESTED** [review]
Please check the comments

**Review by ahmad-kemsan — APPROVED** [review]

**Pull request merged**